src/table/internal.tsx

```diff
@@ -188,6 +189,17 @@ const InternalTable = React.forwardRef(
       [loading]
     );
 
+    const analyticsMetadata = getAnalyticsMetadataProps(rest);
+
+    const { setLastUserAction } = useTableInteractionMetrics({
+      loading,
+      instanceIdentifier: analyticsMetadata?.instanceIdentifier,
+      itemCount: items.length,
+      getComponentIdentifier: () =>
+        toolsHeaderPerformanceMarkRef.current?.querySelector<HTMLElement>(`.${headerStyles['heading-text']}`)
+          ?.innerText ?? toolsHeaderPerformanceMarkRef.current?.innerText,
```

Comment:
We already have this code instance

https://github.com/cloudscape-design/components/blob/da7e1c13faee03f76990bac93afdaf73bf58ac04/src/table/internal.tsx#L180-L181

How about reusing this somehow